scipy/sparse/_construct.py

```diff
@@ -850,6 +850,18 @@ def random(m, n, density=0.01, format='coo', dtype=None,
     """Generate a sparse matrix of the given shape and density with randomly
     distributed values.
 
+    .. warning::
+
+        Since numpy 1.17, passing ``np.random.Generator`` for ``random_state``
+        will lead to much faster execution times. For example:
+
+        >>> from scipy.sparse import random
+        >>> from numpy.random import default_rng
+        >>> random(100_000, 100_000, density=0.001, random_state=default_rng(0))
```

Comment:
I think this positive example of using `default_rng()` is adequately documented in the `Examples` section. Rather, we should warn _against_ omitting the `random_state` argument or passing other values (which will use the slower `RandomState.choice()` implementation).

Reply:
I kept the example in the examples section. However, I don't think it's super clear that the example in the example section conferred any benefit by passing `default_rng`, so I've added some prose explaining that. I do think it's worth being very explicit about this since there are three ways to do this wrong:

* Do nothing
* Pass an integer
* Pass an instance of the class the argument is named after